user/models.py

```diff
@@ -48,24 +37,35 @@ def role(self):
         else:
             return "Contributor"
 
+    # We write a setter as we construct temporary users when doing password validation, and sometimes role information
+    # is part of the data.
     @role.setter
     def role(self, value):
-        self.__roles_cache = set(value or self.groups.values_list("name", flat=True))
-
-    def get_all_roles(self):
-        if self.__roles_cache is None:
-            self.role = None  # force cache to update
-        return self.__roles_cache
+        pass
 
     def at_least(self, role):
-        return role in self.get_all_roles()
+        """Returns if a user has at least a given role's privileges"""
+        return self.is_staff or role == self.role or role in GROUPS.get(self.role, [])
+
+    @property
+    def is_editor(self):
+        """Is this user an editor?"""
+        warnings.simplefilter("always", DeprecationWarning)
+        warnings.warn(
+            "Deprecated usage of SluglineUser::is_editor --- use SluglineUser::role instead",
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        warnings.simplefilter("default", DeprecationWarning)
+        return self.at_least("Editor")
 
     class Meta:
         ordering = ["date_joined"]
 
 
 class UserSerializer(serializers.ModelSerializer):
-    is_editor = serializers.BooleanField(default=False)
+    is_editor = serializers.ReadOnlyField(default=False)
```

Comment:
Are we still intending to use this field? I don't see it being set anywhere.

Reply:
We still need it as the front-end depends on it, and I haven't had the time to implement the front-end portion of this PR yet.

Reply:
So I got the front-end: 19782bb